File: Python version/base/InputProgram.py
class InputProgram(object):
    """Represents a generic option"""
    
    def __init__(self):
        """Creates a new programs , setting space as default separator"""
        self._programs = ""  # Where programs data is stored
        self.__files_paths = list()  # Where associated files are stored
        self._separator = " "  # Used as separator for programs


    def addObjectInput(self,inputObj):
        raise "Functionality not implemented"

    def addObjectsInput(self, inputObjs):
        for inputObj in inputObjs:
            self.addObjectInput(inputObj)

    def addFilesPath(self, file_path):
        """Add a new file path into __files_paths
        The parameter file_path represents a new file path
        """
        self.__files_paths.append(file_path)


    def addProgram(self, new_instruction):
        """Adds a new instruction directly into _programs
        The parameter new_instruction rapresents a new programs instruction
        """
        if(self._programs == None):
            self._programs = new_instruction
        else:
            self._programs += self._separator + new_instruction

    def clearFilesPaths(self):
        """After this method, __files_paths will be empty"""
        del self.__files_paths[:]

    def clearPrograms(self):
        """After this method, _programs will be empty"""
        self._programs = None

    def clearAll(self):
        """After this method, both __files_paths and _programs will be empty"""
        self.clearFilesPaths()
        self.clearPrograms()

    def getFilesPaths(self):
        """Returns the __files_paths list"""
        return self.__files_paths

    def getPrograms(self):
        """Returns data stored in _programs"""
        return self._programs

    def getSeparator(self):
        """Separator the _separator character"""
        return self._separator

    def getStringOfFilesPaths(self):
        """Returns string concatenating files paths"""
        to_return=""
        for paths in self.__files_paths:
            if(len(paths) != 0):
                to_return += paths + " "
        return to_return

    def setPrograms(self, programs):
        """Sets _programs value to the new given one
        The parameter programs, rapresents new value
        """
        self._programs = programs

    def setSeparator(self, separator):
        """Set programs separator to current value
        The parameter separator is used as new separator
        """
        self._separator = separator

File: CSharp version/EmbASP/it/unical/mat/embasp/base/InputProgram.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace it.unical.mat.embasp.@base
{


	public class InputProgram
	{
		protected internal string programs;
		private IList<string> files_paths;
		protected internal string separator;

		public InputProgram()
		{
			init();
			separator = " ";
		}

		public InputProgram(object inputObj)
		{
			init();
			addObjectInput(inputObj);
		}

		public InputProgram(string initial_program)
		{
			init();
			programs = initial_program;
		}

		public virtual void addFilesPath(string file_path)
		{
			files_paths.Add(file_path);
		}

		public virtual void addObjectInput(object inputObj)
		{
			throw new System.NotSupportedException("functionality not implemented");
		}

		public virtual void addObjectsInput(ISet<object> inputObjs)
		{
			foreach (Object inputObj in inputObjs)
			{
				addObjectInput(inputObj);
			}
		}

		public virtual void addProgram(string new_instruction)
		{
			if (programs.Length == 0)
			{
				programs = new_instruction;
			}
			else
			{
				programs += separator + new_instruction;
			}
		}

		public virtual void clearAll()
		{
			clearFilesPaths();
			clearPrograms();
		}

		public virtual void clearFilesPaths()
		{
			files_paths.Clear();
		}

		public virtual void clearPrograms()
		{
			programs = "";
		}

		public virtual IList<string> FilesPaths
		{
			get
			{
				return files_paths;
			}
		}

		public virtual string Programs
		{
			get
			{
				return programs;
			}
			set
			{
				this.programs = value;
			}
		}

		public virtual string Separator
		{
			get
			{
				return separator;
			}
			set
			{
				this.separator = value;
			}
		}

		public virtual string StringOfFilesPaths
		{
			get
			{
    
				StringBuilder to_return = new StringBuilder();
    
				foreach (String paths in files_paths)
				{
					if (paths.Length != 0)
					{
						to_return.Append(paths).Append(" ");
					}
				}
    
				return to_return.ToString();
			}
		}

		private void init()
		{
			programs = "";
			files_paths = new List<string>();
		}



	}

}